extensions/amp-analytics/0.1/events.js

```diff
@@ -168,20 +198,28 @@ export class CustomEventTracker extends EventTracker {
    * @param {!AnalyticsEvent} event
    */
   trigger(event) {
-    // Buffer still exists - enqueue.
-    if (this.buffer_) {
-      let buffer = this.buffer_[event.type];
-      if (!buffer) {
-        buffer = [];
-        this.buffer_[event.type] = buffer;
+    const eventType = event.type;
+    const isSandboxEvent = startsWith(eventType, 'sandbox-');
+    const observables = this.observables_[eventType];
+
+    // If listeners already present - trigger right away.
+    if (observables) {
+      observables.fire(event);
+      if (isSandboxEvent) {
+        // No need to buffer sandbox event if handler ready
+        return;
       }
-      buffer.push(event);
     }
 
-    // If listeners already present - trigger right away.
-    const observers = this.observers_[event.type];
-    if (observers) {
-      observers.fire(event);
+    // Create buffer and enqueue buffer if needed
+    if (isSandboxEvent) {
+      this.sandboxBuffer_[eventType] = this.sandboxBuffer_[eventType] || [];
+      this.sandboxBuffer_[eventType].push(event);
+    } else {
+      if (this.buffer_) {
```

Comment:
I can, but then I need an extra param to store whether `this.buffer_` has expired or not. It would still require me to check here I think.